diff --git a/build b/build
index 6fb680ae..5468f72d 100755
--- a/build
+++ b/build
@@ -34,10 +34,14 @@ try {
 var options = {cwd: __dirname, stdio: 'inherit'};
 var full = '';
 
+var newargv = []
+if(process.argv.includes('--no-update')) newargv.push('--no-update')
+if(process.argv.includes('--no-rebuild')) newargv.push('--no-rebuild')
+
 switch (process.argv[2] || '') {
 case 'full':
 	execFileSync(process.execPath, ['./build-tools/build-indexes'].concat(
-		process.argv.includes('--no-update') ? ['--no-update'] : []
+			newargv
 	), options);
 	execSync(`node ./build-tools/build-learnsets`, options);
 	execSync(`node ./build-tools/build-minidex`, options);
diff --git a/build-tools/build-indexes b/build-tools/build-indexes
index e40dd3d4..4ad61379 100755
--- a/build-tools/build-indexes
+++ b/build-tools/build-indexes
@@ -9,22 +9,27 @@ process.chdir(__dirname + '/..');
 fs.mkdirSync('play.pokemonshowdown.com/data', { recursive: true });
 
 const noUpdate = process.argv.includes('--no-update');
-if (!fs.existsSync('caches/pokemon-showdown')) {
-	if (noUpdate) {
-		console.error('Missing caches/pokemon-showdown. Check out the server before using --no-update.');
-		process.exit(1);
-	}
-	child_process.execSync('git clone https://github.com/smogon/pokemon-showdown.git', {
-		cwd: 'caches',
-	});
-}
+const noRebuild = process.argv.includes('--no-rebuild');
+if(!noRebuild) {
+		if (!fs.existsSync('caches/pokemon-showdown')) {
+			if (noUpdate) {
+				console.error('Missing caches/pokemon-showdown. Check out the server before using --no-update.');
+				process.exit(1);
+			}
+			child_process.execSync('git clone https://github.com/WilliamBundy/pokemon-showdown.git', {
+				cwd: 'caches',
+			});
+		}
 
-process.stdout.write(
-	noUpdate ? "Building data from existing server checkout... " : "Syncing data from Git repository... "
-);
-if (!noUpdate) child_process.execSync('git pull', { cwd: 'caches/pokemon-showdown' });
-child_process.execSync('npm run build', { cwd: 'caches/pokemon-showdown' });
-console.log("DONE");
+		process.stdout.write(
+			noUpdate ? "Building data from existing server checkout... " : "Syncing data from Git repository... "
+		);
+		if (!noUpdate) child_process.execSync('git pull', { cwd: 'caches/pokemon-showdown' });
+		if (!noRebuild) child_process.execSync('npm run build', { cwd: 'caches/pokemon-showdown' });
+		console.log("DONE");
+} else {
+		console.log('SKIPPING');
+}
 
 const Dex = require('../caches/pokemon-showdown/dist/sim/dex').Dex;
 const CompoundWordNames = require('../caches/pokemon-showdown/dist/data/aliases').CompoundWordNames;
@@ -275,6 +280,7 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 		8.6, 8.4, 8.2, -8.4, -8.6,
 		7.1,
 		5.1,
+		4.001, -4.001, //gen4legacy, gen4legacydoubles
 		3.1, 3.11,
 	];
 
@@ -300,6 +306,7 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 		const isBW1 = genIdent === 5.1;
 		const isRS = genIdent === 3.1;
 		const isFRLG = genIdent === 3.11;
+		const isLegacy = ('' + genIdent).endsWith('.001')
 		const gen = (() => {
 			let genStr = 'gen' + genNum;
 			if (isSSDLC1) genStr += 'dlc1';
@@ -313,9 +320,10 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 			if (isRS) genStr += 'rs';
 			if (isFRLG) genStr += 'frlg';
 			if (isChampions || isNDChampions) genStr = 'champions';
+			if (isLegacy) genStr += 'legacy';
 			return genStr;
 		})();
-		// process.stdout.write("" + gen + (isDoubles ? " doubles" : "") + "... ");
+		process.stdout.write("" + gen + (isDoubles ? " doubles" : "") + "... \n");
 		const pokemon = Object.keys(Dex.data.Pokedex);
 		pokemon.sort();
 		const tierTable = {};
@@ -564,12 +572,16 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 			BattleTeambuilderTable[gen + 'vgc'].formatSlices = formatSlices;
 			if (genNum === 9) BattleTeambuilderTable[gen + 'vgc'].metagameBans = metagameBans;
 		} else if (isDoubles) {
+			if(isLegacy) console.log("adding " + gen + " in doubles");
 			BattleTeambuilderTable[gen + 'doubles'] = {};
 			BattleTeambuilderTable[gen + 'doubles'].tiers = tiers;
 			BattleTeambuilderTable[gen + 'doubles'].items = items;
 			BattleTeambuilderTable[gen + 'doubles'].overrideTier = overrideTier;
 			BattleTeambuilderTable[gen + 'doubles'].metagameBans = metagameBans;
 			BattleTeambuilderTable[gen + 'doubles'].formatSlices = formatSlices;
+			if(isLegacy) {
+				BattleTeambuilderTable[gen+'doubles'].learnsets = {};
+			}
 		} else if (isGen9BH) {
 			BattleTeambuilderTable['bh'] = {};
 			BattleTeambuilderTable['bh'].tiers = tiers;
@@ -596,6 +608,7 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 			BattleTeambuilderTable[gen].nonstandardMoves = nonstandardMoves;
 			BattleTeambuilderTable[gen].learnsets = {};
 		} else {
+			if(isLegacy) console.log("adding " + gen + " in final else");
 			BattleTeambuilderTable[gen] = {};
 			BattleTeambuilderTable[gen].overrideTier = overrideTier;
 			BattleTeambuilderTable[gen].tiers = tiers;
@@ -608,6 +621,10 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 				BattleTeambuilderTable[gen].nonstandardMoves = nonstandardMoves;
 				BattleTeambuilderTable[gen].learnsets = {};
 			}
+			if(isLegacy) {
+				BattleTeambuilderTable[gen].learnsets = {};
+			}
+
 		}
 
 		const tierOrder = (() => {
@@ -1209,6 +1226,24 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 		}
 	}
 
+	// NOTE(will): Based on Casper's code found here, big thanks
+	// https://github.com/Casper1123/pokemon-showdown-client/blob/3a31e9c38b9e200722aed94912d1cf9ff2256b1b/build-tools/build-indexes#L809
+	const G4LegacyLearnsets = Dex.mod('gen4legacy').data.Learnsets;
+	for (const tableId of ['gen4legacy', 'gen4legacydoubles']) {
+		for (const id in G4LegacyLearnsets) {
+			const species = Dex.mod('gen4legacy').species.get(id);
+			if (species.isNonstandard && ['Unobtainable', 'CAP'].includes(species.isNonstandard)) continue;
+
+			// NOTE(will): this is a bit of a hack that I can get away with because 
+			// in my mod, all the movesets are implemented as "learned at level 1 in this generation"
+			// and I'm relying on that because I don't know the syntax properly
+			BattleTeambuilderTable[tableId].learnsets[id] = {};
+			for(const moveid in G4LegacyLearnsets[id].learnset) {
+				BattleTeambuilderTable[tableId].learnsets[id][moveid] = '4';
+			}
+		}
+	}
+
 	// Client relevant data that should be overridden by past gens and mods
 	const overrideSpeciesKeys = [
 		'abilities', 'baseStats', 'cosmeticFormes', 'isNonstandard', 'requiredItems', 'types', 'unreleasedHidden',
@@ -1307,7 +1342,7 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 	//
 
 	for (const mod of [
-		'gen3rs', 'gen3frlg', 'gen5bw1', 'gen7letsgo', 'gen8bdsp', 'gen9ssb', 'champions',
+		'gen3rs', 'gen3frlg', 'gen5bw1', 'gen7letsgo', 'gen8bdsp', 'gen9ssb', 'champions', 'gen4legacy', 'gen4legacydoubles'
 	]) {
 		const modDex = Dex.mod(mod);
 		const modData = modDex.data;
diff --git a/play.pokemonshowdown.com/src/battle-dex-search.ts b/play.pokemonshowdown.com/src/battle-dex-search.ts
index e3806460..0cd6fd8e 100644
--- a/play.pokemonshowdown.com/src/battle-dex-search.ts
+++ b/play.pokemonshowdown.com/src/battle-dex-search.ts
@@ -674,7 +674,7 @@ abstract class BattleTypedSearch<T extends SearchType> {
 
 	protected formatType: 'doubles' | 'bdsp' | 'bdspdoubles' | 'rs' | 'frlg' | 'bw1' | 'letsgo' | 'metronome' | 'natdex' |
 		'nfe' | 'ssdlc1' | 'ssdlc1doubles' | 'predlc' | 'predlcdoubles' | 'svdlc1' | 'svdlc1doubles' | 'stadium' | 'lc' |
-		'champions' | 'natdexchampions' | null = null;
+		'champions' | 'natdexchampions' | 'legacy' | 'legacydoubles' | null = null;
 	isDoubles = false;
 
 	/**
@@ -698,6 +698,8 @@ abstract class BattleTypedSearch<T extends SearchType> {
 		this.baseResults = null;
 		this.baseIllegalResults = null;
 
+		let originalFormat = format;
+
 		if (format.startsWith('gen')) {
 			const gen = (Number(format.charAt(3)) || 6);
 			format = (format.slice(4) || 'customgame') as ID;
@@ -706,6 +708,22 @@ abstract class BattleTypedSearch<T extends SearchType> {
 			this.dex = Dex;
 		}
 
+		// NOTE(will) fix for legacy mods
+		// FIXME: I hate this, I think all the format/mod information 
+		// is available to the client here, but I guess it insists on doing this?
+		// why? why why why why why?
+		// hopefully we don't end up with any name collisions, but who knows!
+		if(format.startsWith('ptl') || format.startsWith('el') || format.startWith('xyl') || format.startsWith('cl') || format.startsWith('yl')) {
+			this.dex = Dex.mod(originalFormat.slice(0, 4) + 'legacy')
+			this.formatType = 'legacy';
+
+			// hack for older gens, may be uneccessary
+			if (format.includes('doubles')) {
+				this.formatType = 'legacydoubles';
+				this.isDoubles = true;
+			}
+		}
+
 		if (format.startsWith('dlc1') && this.dex.gen === 8) {
 			if (format.includes('doubles')) {
 				this.formatType = 'ssdlc1doubles';
@@ -912,6 +930,12 @@ abstract class BattleTypedSearch<T extends SearchType> {
 		if (this.formatType === 'frlg') table = table['gen3frlg'];
 		if (this.formatType === 'champions') table = table['champions'];
 		if (this.formatType === 'natdexchampions') table = table['natdexchampions'];
+		//FIXME(will): it's another one
+		if(this.formatType == 'legacy') {
+			if(this.dex.gen == 4) {
+				table = table['gen4legacy'];
+			}
+		}
 		if (speciesid in table.learnsets) return speciesid;
 		const species = this.dex.species.get(speciesid);
 		if (!species.exists) return '' as ID;
@@ -1185,6 +1209,11 @@ class BattlePokemonSearch extends BattleTypedSearch<'pokemon'> {
 			}
 		} else if (this.formatType === 'stadium') {
 			table = table[`gen${dex.gen}stadium${dex.gen > 1 ? dex.gen : ''}`];
+		} else if (this.formatType?.startsWith('legacy')) {
+			//FIXME(will): would there even be a point in trying to fix this?
+			if(this.dex.gen == '4') {
+				table = table['gen4legacy'];
+			}
 		}
 
 		if (!table.tierSet) {
@@ -1505,6 +1534,9 @@ class BattleItemSearch extends BattleTypedSearch<'item'> {
 			table = table[`champions`];
 		} else if (this.formatType === 'natdexchampions') {
 			table = table[`natdexchampions`];
+		//FIXME(will): horrible hacks because that's the showdown way apparently
+		} else if(this.formatType?.startsWith('ptl')) {
+			table = table['gen4legacy'];
 		} else if (this.dex.gen < 9) {
 			table = table[`gen${this.dex.gen}`];
 		}
@@ -1897,6 +1929,14 @@ class BattleMoveSearch extends BattleTypedSearch<'move'> {
 		if (this.formatType?.startsWith('ssdlc1')) lsetTable = lsetTable['gen8dlc1'];
 		if (this.formatType?.startsWith('predlc')) lsetTable = lsetTable['gen9predlc'];
 		if (this.formatType?.startsWith('svdlc1')) lsetTable = lsetTable['gen9dlc1'];
+		//FIXME(will) why? why this? why can't we have a global function that transforms
+		// a format into a modid based on, idk, the information provided in formats.ts?????
+		// why is it like this everywhere???
+		if (this.formatType?.includes('legacy')) {
+			if(gen === '4') {
+				lsetTable = lsetTable['gen4legacy'];
+			} 
+		}
 		while (learnsetid) {
 			let learnset = lsetTable.learnsets[learnsetid];
 			if (learnset) {
diff --git a/play.pokemonshowdown.com/src/battle-dex.ts b/play.pokemonshowdown.com/src/battle-dex.ts
index 150ad291..3f78c445 100644
--- a/play.pokemonshowdown.com/src/battle-dex.ts
+++ b/play.pokemonshowdown.com/src/battle-dex.ts
@@ -542,6 +542,14 @@ export const Dex = new class implements ModdedDex {
 		if (dex.gen === 9 && formatid.includes('champions')) {
 			dex = Dex.mod('champions' as ID);
 		}
+
+
+		//FIXME(will): this is a terrible hack for testing
+		if(formatid.startsWith('ptl')) {
+			dex = Dex.mod('gen4legacy');
+		}
+
+		
 		return dex;
 	}
 
diff --git a/play.pokemonshowdown.com/testclient-old.html b/play.pokemonshowdown.com/testclient-old.html
index 024df44e..0115e9fb 100644
--- a/play.pokemonshowdown.com/testclient-old.html
+++ b/play.pokemonshowdown.com/testclient-old.html
@@ -12,7 +12,7 @@
 		<link rel="stylesheet" href="style/font-awesome.css" />
 		<meta name="robots" content="noindex" />
 		<meta http-equiv="X-UA-Compatible" content="IE=Edge" />
-		<script src="https://play.pokemonshowdown.com/config/config.js"></script>
+		<script src="config/config.js"></script>
 		<script>
 			function loadRemoteData(src) {
 				var scriptEl = document.createElement('script');
